test/src/unit-regression2.cpp

```diff
@@ -724,10 +723,11 @@ TEST_CASE("regression tests 2")
         test2[ordered_json::json_pointer(p)] = json::object();
         CHECK(test2.dump() == "{\"root\":{}}");
 
-        // mixed type - the JSON Pointer is implicitly converted into a string "/root"
+        // json::json_pointer and ordered_json::json_pointer are the same type; behave as above
         ordered_json test3;
         test3[json::json_pointer(p)] = json::object();
-        CHECK(test3.dump() == "{\"/root\":{}}");
+        CHECK(std::is_same<json::json_pointer::string_t, ordered_json::json_pointer::string_t>::value);
```

Comment:
I should add similar checks to the backwards compatibility section in `unit-json_pointer.cpp`.